tests/console/generate_dud.pm

```diff
@@ -13,23 +13,75 @@ use strict;
 use warnings;
 use base "opensusebasetest";
 use testapi;
-use utils qw(zypper_call);
+use XML::Writer;
+use util qw(zypper_call);
 use autoyast qw(expand_variables);
 use registration qw(add_suseconnect_product get_addon_fullname);
 
+sub generate_xml {
+    my $addons = get_var('ADD_ON_PRODUCTS', '');
+
+    # Generate addon products xml file
+    my $writer = XML::Writer->new(
+        DATA_MODE => 'true',
+        DATA_INDENT => 2,
+        OUTPUT => "self"
+    );
+    $writer->startTag(
+        "add_on_products",
+        xmlns => "http://www.suse.com/1.0/yast2ns",
+        "xmlns:config" => "http://www.suse.com/1.0/configns"
+    );
+    $writer->startTag("product_items", "config:type" => "list");
+    for my $addon (split(/,/, $addons)) {
+        my @n = split(/\//, $addon);
+        my $name = $n[-1] . "_" . $n[-2];
+        $writer->startTag("product_item");
+        $writer->startTag("url");
+        $writer->characters($addon);
+        $writer->endTag("url");
+        $writer->startTag("name");
+        $writer->characters($name);
+        $writer->endTag("name");
+        $writer->startTag("alias");
+        $writer->characters($name);
+        $writer->endTag("alias");
+        $writer->startTag("priority", "config:type" => "integer");
+        $writer->characters("50");
+        $writer->endTag("priority");
+        $writer->startTag("ask_user", "config:type" => "boolean");
+        $writer->characters("true");
+        $writer->endTag("ask_user");
+        $writer->startTag("selected", "config:type" => "boolean");
+        $writer->characters("true");
+        $writer->endTag("selected");
+        $writer->startTag("check_name", "config:type" => "boolean");
+        $writer->characters("true");
+        $writer->endTag("check_name");
+        $writer->endTag("product_item");
+    }
+    $writer->endTag("product_items");
+    $writer->endTag("add_on_products");
+    $writer->end();
+    return $writer->to_string;
+}
+
 sub run {
     select_console 'root-console';
 
     my $xml = 'add_on_products.xml';
     my $dud = get_required_var('DUD');
 
-    my $content = expand_variables(get_test_data($xml));
+    my $content =
+      (get_var('ADD_ON_PRODUCTS', '') eq '')
+      ? expand_variables(get_test_data($xml))
+      : generate_xml;
     save_tmp_file($xml, $content);
     add_suseconnect_product(get_addon_fullname('phub'));
     zypper_call('in mkdud');
     assert_script_run('wget -P inst-sys ' . autoinst_url . "/files/$xml");
-    assert_script_run("mkdud --create $dud --dist sle15 " .
-          "--install instsys,repo --obs-keys --name 'Update' inst-sys");
+    assert_script_run("mkdud --create $dud --dist sle15 "
+          . "--install instsys,repo --obs-keys --name 'Update' inst-sys");
     upload_asset($dud);
```

Comment:
we should upload the generated xml file, no need to pass it to the child, but for being able to verify correct xml.